packages/react/src/components/GroupedList/GroupedListV2.base.tsx

```diff
@@ -0,0 +1,604 @@
+  return ev.which === getRTLSafeKeyCode(KeyCodes.right);
+};
+
+const getClassNames = classNamesFunction<IGroupedListStyleProps, IGroupedListStyles>();
+
+const getKey: IListProps['getKey'] = (item, _index) => {
+  switch (item.type) {
+    case 'item':
+      return item.item?.key ?? null;
+
+    case 'header':
+      return item.group.key;
+
+    case 'footer':
+      return `${item.group.key}-footer`;
+
+    case 'showAll':
+      return `${item.group.key}-showAll`;
+  }
+
+  return null;
+};
+
+const renderGroupHeader = (props: IGroupHeaderProps): JSX.Element => {
+  return <GroupHeader {...props} />;
+};
+
+const renderGroupShowAll = (props: IGroupShowAllProps): JSX.Element => {
+  return <GroupShowAll {...props} />;
+};
+
+const renderGroupFooter = (props: IGroupFooterProps): JSX.Element | null => {
+  if (props.group && props.footerText) {
+    return <GroupFooter {...props} />;
+  }
+
+  return null;
+};
+
+export const GroupedListV2FC: React.FC<IGroupedListV2Props> = props => {
+  const {
+    selection,
+    selectionMode = SelectionMode.multiple,
+    groupProps = {},
+    compact = false,
+    items = [],
+    groups,
+    onGroupExpandStateChanged,
+
+    className,
+    usePageCache,
+    onShouldVirtualize,
+    theme,
+    role = 'treegrid',
+    styles,
+    focusZoneProps = {},
+    rootListProps = {},
+    onRenderCell,
+    viewport,
+    listRef,
+    groupExpandedVersion,
+  } = props;
+
+  const {
+    onRenderHeader = renderGroupHeader,
+    onRenderFooter = renderGroupFooter,
+    onRenderShowAll = renderGroupShowAll,
+  } = groupProps;
+
+  const classNames: IProcessedStyleSet<IGroupedListStyles> = getClassNames(styles, {
+    theme: theme!,
+    className,
+    compact,
+  });
+
+  const events = React.useRef<EventGroup>();
+  const flatList = React.useRef<IGroupedItem[]>([]);
+  const isSomeGroupExpanded = React.useRef<boolean>(computeIsSomeGroupExpanded(groups));
+
+  const [version, setVersion] = React.useState({});
+  const [toggleVersion, setToggleVersion] = React.useState({});
+
+  // eslint-disable-next-line deprecation/deprecation
+  const { shouldEnterInnerZone = isInnerZoneKeystroke } = focusZoneProps;
+
+  const listView = React.useMemo(() => {
+    return flattenItems(groups, items, flatList.current, groupProps);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [groups, groupProps, items, toggleVersion, flatList]);
+
+  const getPageSpecification = React.useCallback(
+    (flattenedIndex: number): { key?: string } => {
+      const pageGroup = listView[flattenedIndex];
+      return {
+        key: pageGroup.type === 'header' ? pageGroup.group.key : undefined,
+      };
+    },
+    [listView],
+  );
+
+  useMount(() => {
+    if (groupProps?.isAllGroupsCollapsed) {
+      setGroupsCollapsedState(groups, groupProps.isAllGroupsCollapsed);
+    }
+    events.current = new EventGroup(this);
+  });
+
+  useUnmount(() => {
+    events.current?.dispose();
+  });
```

Comment:
Agreed and I've made this change as it will also remove all the `useMount()` and `useUnmount()` code from this component. However, one of the downsides is that the `useEffect()` version has to specify:

`// eslint-disable-next-line react-hooks/exhaustive-deps`

because technically the effect has those dependencies, even though we want to ignore them. Not the end of the world but I typically look at that sort of thing as a code smell.